codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -337,6 +337,7 @@ private void sendPreface(ChannelHandlerContext ctx) {
             if (!connection().isServer()) {
                 // Clients must send the preface string as the first bytes on the connection.
                 ctx.write(connectionPrefaceBuf()).addListener(ChannelFutureListener.CLOSE_ON_FAILURE);
+                ctx.fireUserEventTriggered(Http2ConnectionPrefaceWrittenEvent.INSTANCE);
```

Comment:
@normanmaurer - This is currently not necessary. If the transport is ordered (currently required for HTTP/2) then it is good enough to know we attempted the write. We shouldn't have to wait for a flush call.